WcaOnRails/app/models/user.rb

```diff
@@ -293,6 +285,49 @@ def after_confirmation
     end
   end
 
+  def software_team?
+    team_member?('software')
+  end
+
+  def results_team?
+    team_member?('results')
+  end
+
+  def wrc_team?
+    team_member?('wrc')
+  end
+
+  def wdc_team?
+    team_member?('wdc')
+  end
+
+  def team_member?(team)
+    self.team_members.where(team_id: ( self.teams.find_by_friendly_id(team) ) ).each do |member|
+      if member.current_member?
+        return true
+      end
+    end
+    return false
+  end
+
+  def was_team_member?(team)
```

Comment:
Well, since there will be just a few people editing this kind of stuff, I hope we can educate them as to how it works :)
A team member would never be deleted, unless there was a mistake. Past members should have the end date set (even if it doesn't reflect reality 100%).